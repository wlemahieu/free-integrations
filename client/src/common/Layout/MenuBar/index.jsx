import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Icon, Menu } from 'antd';
import getSelectedKeys from 'utilities/getLocationPaths';

const MenuBar = props => {
  const selectedKeys = getSelectedKeys(props.location);
  return (
    <Menu
      className="menu"
      theme="dark"
      mode="horizontal"
      defaultSelectedKeys={['/']}
      selectedKeys={[selectedKeys]}
    >
      <Menu.Item key="/">
        <Link to="/">Home</Link>
      </Menu.Item>
      <Menu.Item key="/chat">
        <Link to="/chat">Chat</Link>
      </Menu.Item>
      <Menu.Item key="/news">
        <Link to="/news">News</Link>
      </Menu.Item>
      <Menu.Item style={{ float: 'right' }}>
        <Icon type="question-circle" />
      </Menu.Item>
    </Menu>
  );
};

MenuBar.propTypes = {
  location: PropTypes.object
};

MenuBar.defaultProps = {
  location: {}
};

export default MenuBar;
