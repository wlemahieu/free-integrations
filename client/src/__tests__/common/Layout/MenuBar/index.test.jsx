import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import MenuBar from 'common/Layout/Foot';

let wrapper;
beforeEach(() => {
  wrapper = render(
    <Router>
      <MenuBar location={{}} />;
    </Router>
  );
});
afterEach(() => {
  wrapper = undefined;
});

describe('Layout Foot', () => {
  it('should match snapshot', () => {
    const { container } = wrapper;
    expect(container).toMatchSnapshot();
  });
});
