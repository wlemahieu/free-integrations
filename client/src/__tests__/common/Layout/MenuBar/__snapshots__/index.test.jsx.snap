// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Foot should match snapshot 1`] = `
<div>
  <footer
    class="centered ant-layout-footer"
  >
    <span>
      Free Integrations
    </span>
    <a
      href="/license"
      title="License"
    >
       (MIT License) 
    </a>
    <span>
      Created by Wesley LeMahieu
    </span>
  </footer>
  ;
</div>
`;
